tests/unit/imutils/test_exif.py

```diff
@@ -11,17 +11,36 @@
 from vimiv.imutils import exif
```

Comment:
I need some more time to think about the testing in general, so this review will not cover the tests yet :smile:

Reply:
If you don't mind, I feel like it would be easier for both of us to try to get the CI passing here and merge this PR afterwards. I will likely have to play around a lot to see how to test all three scenarios (py3exiv2, piexif, no exif) with hopefully some smart utilization of pytest fixture parametrization. Unless you have already thought of something, I would just work on it after this PR in some playground branch and ask you for a review.

To get the CI passing I *think* we have to add the actual exiv2 library to the installed dependencies in `.github/workflows/ci.yml`. There are two usages of `apt-get install`, one for the python3.6/3.7 job, one for the python3.8 job. From a quick [package search](https://packages.ubuntu.com/search?keywords=exiv2&searchon=names&suite=bionic&section=all) I hope just adding `exiv2` fixes the problem, but I always have issues with ubuntu packaging :laughing: Could you either try this out or just add the correct packages if you happen to know what is needed on ubuntu?

Reply:
I have tried to get this to works yesterday for quite some time. All required packages should get installed. The installation of `py3exiv2` fails with `/usr/bin/ld: cannot find -lboost_python38`. According to the internet (e.g. [here](https://stackoverflow.com/questions/59437794/usr-bin-ld-cannot-find-lboost-python-py34) ) a simple symlink should fix it. But I have not had any success yet.

Reply:
Thanks for taking multiple attempts at fixing this, it seems like this is much trickier than I had hoped. Not a fan of ubuntu packaging in general, but this in combination with doing it on CI seems horrible :worried:

In case you don't find the magic solution by then, I will start a test project to play around with starting on the 2nd of January, I still have some other stuff to do (including celebrating :laughing:) until then. Would you mind if I merge this into a non-master branch first when and if I find some solution to this havoc so I can continue testing before merging into master?

Reply:
I am sorry that I have not manages to get it running yet :blush: I have tried everything I could think of as well as everything I found online, but without any success. And to be honest I am  also not sure what else to try.

The compilation step `g++ -pthread -shared -Wl,--rpath=/opt/hostedtoolcache/Python/3.8.6/x64/lib -Wl,--rpath=/opt/hostedtoolcache/Python/3.8.6/x64/lib build/temp.linux-x86_64-3.8/src/exiv2wrapper.o build/temp.linux-x86_64-3.8/src/exiv2wrapper_python.o -L/opt/hostedtoolcache/Python/3.8.6/x64/lib -lboost_python38 -lexiv2 -o build/lib.linux-x86_64-3.8/libexiv2python.cpython-38-x86_64-linux-gnu.so` fails with `/usr/bin/ld: cannot find -lboost_python38`.

I am not at all experienced with C++ compiling (only little C compiling). So, for someone more versed the problem may be obvious. Here things I have tried, thoughts and open questions:
- All required packages should be installed (acording to the [official site](https://pypi.org/project/py3exiv2/) as well as this [SO post](https://stackoverflow.com/a/50381827/5464989))
- Are the packages somehow installed in a wrong order?
- Are some packages conflicting?
- `libboost_python-py38.so` is not available in the image, only the python 3, 3.6, 2 and 2.7 version. So I have symlinked one for version 3.8.
- I have created symlinks at different locations (I not really sure where the compiler/linker looks for them): `/usr/lib/x86_64-linux-gnu`, `/usr/lib/` as well as `/opt/hostedtoolcache/Python/3.8.6/x86/lib` (which is explicitly added as a library source in the `g++` command).
- After creating the symlinks does the linker have to be notified in some way (about the new libraries)? I have tried running `ldconf` after creating the links.

The cause of the problem seems to be somehow related to the ubuntu:latest image of the github CI. When conducting the same steps on docker running ubuntu:latest, everything works as excepted (even without manually linking any libraries).

Since I do not think that I am able to resolve this, I will cleanup the commits in this branch. Then you can can continue experiementing and testing as you want.

Reply:
Awesome, thanks a lot once again for this tremendous amount of work. I have merged this into `additional-exiv2` for now so I can force-push, rebase and so forth without all the noise it would make here :blush: